proposals/vector_to_regular_points.json

```diff
@@ -23,16 +23,16 @@
             ]
         },
         {
-            "name": "cell_size",
-            "description": "The size in degrees of the cell of the regular grid. If the cell size is bigger than the geometry, a single point is returned located in the centroid of the geometry. If the centroid is not enclosed in the geometry area, no point is returned.",
+            "name": "distance",
+            "description": "Defines the minimum distance in degrees required between two samples generated for a single geometry.\n\n- For **polygons**, the distance defines the cell sizes of a regular grid that starts at the upper-left bound of each polygon. The centroid of each cell is then a sample point. If the centroid is not enclosed in the polygon, no point is sampled, which results in an empty geometry.\n- For **lines** (line strings), the sampling starts with a point at the first coordinate of the line and then walks along the line and samples a new point each time the distance to the previous point has been reached again.\n- For **points**, the point is returned as given.",
```

Comment:
Basically we don't allow to sample points from geometries with centroid which is not within the geometry? This could be bad if we are using a layer like Corine where all the geometries are pretty complex.

Reply:
This is coming from your proposal above ("If the centroid is not enclosed in the geometry area, no point is returned."). So why have you added it? ;-) We can also specify a different behavior...

Reply:
My proposal was to return the centroid if we can't sample points from the geometry due to its limited size (distance greater than the geometry "size" in lat/lon). If in this case the centroid is not contained within the geometry return an empty one.
From the current description the behavior I understand is: firstly check if the centroid is within the geometry and if yes do the rest, otherwise return an empty feature.

Reply:
Ah! Yes, there's an ambiguity in the description, indeed. I've tries to fix that in the last commit: bcefb4ba
That should work, right?

Reply:
So to clarify, what I assume we are trying to express here: 

![image](https://user-images.githubusercontent.com/8262166/155354383-657151e0-c1c6-4379-b28d-b6b1f0754fd9.png)